app/models/member.rb

```diff
@@ -35,14 +35,16 @@ class Member < ActiveRecord::Base
   enum donor_status: [:nondonor, :donor, :recurring_donor]
 
   def self.find_from_request(akid: nil, id: nil)
-    actionkit_user_id = AkidParser.parse(akid, Settings.action_kit.akid_secret)[:actionkit_user_id]
+    member = find_by_akid(akid)
+    return member if member.present?
+    id.present? ? find_by(id: id) : nil
```

Comment:
if `akid` is blank, then `member` is nil and the `find_by(id` is used. This code behaves just like the previous version. Anyway, this is *veryyy* nasty code, was trying to preserve it just to extract the `find_by_akid` method. But I do think it should be rewritten from scratch, and maybe renamed: the model shouldn't know about requests.